src/Utils/Github.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Utils.Github
+  ( showGithubException
+  , githubAPILimitErrorText
+  )
+where
+
+import Data.Aeson (FromJSON, ToJSON, decode)
+import Data.ByteString.Char8 (ByteString)
+import Data.ByteString.Lazy (fromStrict)
+import Data.Text (Text, isPrefixOf)
+import GHC.Generics (Generic)
+import qualified Network.HTTP.Client as Client
+import qualified Network.HTTP.Req as Req
+import PyF (fmt)
+
+import Utils.Req (showHTTPException, showRawResponse)
+
+-- | Represents a typical Github Error serialized as JSON like so:
+--
+-- @
+-- {
+--    "message": "the error reason"
+-- }
+-- @
+data GithubError = GithubError
+  { message :: Text
+  } deriving (Generic, Show)
+
+instance ToJSON GithubError where
+    -- No need to provide a toJSON implementation.
+
+instance FromJSON GithubError
+    -- No need to provide a parseJSON implementation.
```

Comment:
Can't you just add that in the `deriving` clause of the `GithubError` type?

Reply:
obviously I could if I had the reflex ;)